lib/interfaces/include/BMSDriverGroup.h

```diff
@@ -0,0 +1,259 @@
+ *
+ * NOTE: Because the broadcast command will provide 8 bytes of data per command call from EACH IC,
+ * Each instance LTC6811 will represent functionality for all ICs on a particular chip select
+ * Therefore, the data stored in one instance will only hold HALF. One for chip_select 9 and the other for 10
+ * During instantiation, we need to make sure that the addresses are assigned for each LTC6811
+ */
+
+/* Library Includes */
+/**
+ * INCLUDE: LTC6811-1.h and LTC6811-2.h for model-specific data packaging
+ * LTC6811-1 can only function through broadcasted commands and therefore used daisy-chained
+ * LTC6811-2 can operate under broadcast and address functionality, but will only perform address-based commands
+ * Irregardless of which model we choose to use, we need will need access to Arduino and SPI repositories.
+ */
+#include <Arduino.h>
+#include <SPI.h>
+// #include <string.h>
+#include <stdio.h>
+#include "Configuration.h"
+#include "LTCSPIInterface.h"
+#include <cstdint>
+#include <optional>
+
+template <size_t num_chips, size_t num_chip_selects>
+class BMSDriverGroup
+{
+public:
+    using volt = float;
+    using celcius = float; 
+    struct BMSData
+    {
+        std::array<std::array<std::optional<volt>, 12>, num_chips> voltages;
+        std::array<celcius, 4 * num_chips> cell_temperatures;
+        std::array<float, (num_chips + 1) / 2> humidity;
+        std::array<float, (num_chips + 1) / 2> board_temperatures;
+        float min_voltage;
+        float max_voltage;
+        size_t min_voltage_cell_id; // 0 - 125
+        size_t max_voltage_cell_id; // 0 - 125
+        size_t max_board_temperature_segment_id; // 0 - 5
+        size_t max_humidity_segment_id; // 0 - 5
+        size_t max_cell_temperature_cell_id; // 0 - 47
+        float total_voltage;
+        float average_cell_temperature;
+    };
+
+    BMSDriverGroup() {};
+
+public:
+    /* -------------------- SETUP FUNCTIONS -------------------- */
+
+    /**
+     * INIT: ONLY CALLED ONCE at initialization. LTC6811 only needs to set up the PEC table
+     * @post We should have initialized the PEC by calling generate_PEC_table(), so every other time
+     * we need to find the PEC it will be much faster.
+     */
+    void init();
+
+    /* -------------------- READING DATA FUNCTIONS -------------------- */
+
+    /**
+     * Reads Voltage and GPIO data from ALL 12 Cell Voltages and 2 Auxillary GPIO Registers on the device.
+     * For each BMS segment, there is 6 board thermistors and 2 humidity sensors.
+     * NOTE: Conversions are different depending on which we are reading.
+     * @pre in order to actually "read" anything, we need to call wakeup() and send data over SPI
+     * @post store all temperature data into the board_temperatures container
+     * AND record the maximum value and locations
+     */
+    // void read_thermistor_and_humidity();
+    BMSData read_data(const std::array<uint16_t, num_chips> &cell_balance_statuses);
```

Comment:
non-ergonomic interface. it is fine to maintain the internal state of the `cell_balancing_status` for this case as it seems that the protocol works best when you send the command for these statuses again, so this puts the onus on the user to maintain this balancing status state, however that should instead be maintained by this class itself and the read_data() should not require any arguments.